### PR TITLE
:tada: Source Amazon Seller Partner: Add new streams

### PR DESCRIPTION
**Pull request by vladimir-remar**
## What
Add two new report types:
- GET_FBA_FULFILLMENT_CUSTOMER_RETURNS_DATA
- GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE
## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

```bash
============================================================================================ test session starts =============================================================================================
platform linux -- Python 3.9.13, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /home/vladimir/.virtualenvs/amzseller/bin/python
cachedir: .pytest_cache
rootdir: /home/vladimir/Work/Airbyte_projects/Fork/airbyte, configfile: pytest.ini
plugins: mock-3.7.0
collected 93 items                                                                                                                                                                                           

unit_tests/test_finance_streams.py::test_finance_stream_next_token PASSED
unit_tests/test_finance_streams.py::test_financial_event_groups_stream_request_params PASSED
unit_tests/test_finance_streams.py::test_financial_event_groups_stream_parse_response PASSED
unit_tests/test_finance_streams.py::test_financial_events_stream_request_params PASSED
unit_tests/test_finance_streams.py::test_financial_events_stream_parse_response PASSED
unit_tests/test_reports_streams_rate_limits.py::test_reports_stream_should_retry PASSED
unit_tests/test_reports_streams_rate_limits.py::test_reports_stream_send_request PASSED
unit_tests/test_reports_streams_rate_limits.py::test_reports_stream_send_request_backoff_exception {"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 5.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: None, Response Code: 429, Response Text: )"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error 'Request URL: None, Response Code: 429, Response Text: ' after 1 tries. Waiting 5 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 10.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: None, Response Code: 429, Response Text: )"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error 'Request URL: None, Response Code: 429, Response Text: ' after 2 tries. Waiting 10 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 20.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: None, Response Code: 429, Response Text: )"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error 'Request URL: None, Response Code: 429, Response Text: ' after 3 tries. Waiting 20 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send_request(...) for 40.0s (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: None, Response Code: 429, Response Text: )"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Caught retryable error 'Request URL: None, Response Code: 429, Response Text: ' after 4 tries. Waiting 40 seconds then retrying..."}}
{"type": "LOG", "log": {"level": "ERROR", "message": "Giving up _send_request(...) after 5 tries (airbyte_cdk.sources.streams.http.exceptions.DefaultBackoffException: Request URL: None, Response Code: 429, Response Text: )"}}
PASSED
unit_tests/test_source.py::test_spec PASSED
unit_tests/test_source.py::test_streams PASSED
unit_tests/test_source.py::test_stream_with_good_iam_arn_value[arn:aws:iam::123456789098:user/some-user] PASSED
unit_tests/test_source.py::test_stream_with_good_iam_arn_value[arn:aws:iam::123456789098:role/some-role] PASSED
unit_tests/test_source.py::test_stream_with_bad_iam_arn_value PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2VIGQ35RCS4UG-input_data0-expected_data0] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2VIGQ35RCS4UG-input_data1-expected_data1] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2VIGQ35RCS4UG-input_data2-expected_data2] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2VIGQ35RCS4UG-input_data3-expected_data3] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1PA6795UKMFR9-input_data4-expected_data4] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1PA6795UKMFR9-input_data5-expected_data5] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1PA6795UKMFR9-input_data6-expected_data6] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1PA6795UKMFR9-input_data7-expected_data7] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1C3SOZRARQ6R3-input_data8-expected_data8] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1C3SOZRARQ6R3-input_data9-expected_data9] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1C3SOZRARQ6R3-input_data10-expected_data10] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1C3SOZRARQ6R3-input_data11-expected_data11] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ARBP9OOSHTCHU-input_data12-expected_data12] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ARBP9OOSHTCHU-input_data13-expected_data13] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ARBP9OOSHTCHU-input_data14-expected_data14] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ARBP9OOSHTCHU-input_data15-expected_data15] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1RKKUPIHCS9HS-input_data16-expected_data16] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1RKKUPIHCS9HS-input_data17-expected_data17] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1RKKUPIHCS9HS-input_data18-expected_data18] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1RKKUPIHCS9HS-input_data19-expected_data19] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A13V1IB3VIYZZH-input_data20-expected_data20] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A13V1IB3VIYZZH-input_data21-expected_data21] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A13V1IB3VIYZZH-input_data22-expected_data22] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A13V1IB3VIYZZH-input_data23-expected_data23] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A21TJRUUN4KGV-input_data24-expected_data24] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A21TJRUUN4KGV-input_data25-expected_data25] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A21TJRUUN4KGV-input_data26-expected_data26] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A21TJRUUN4KGV-input_data27-expected_data27] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[APJ6JRA9NG5V4-input_data28-expected_data28] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[APJ6JRA9NG5V4-input_data29-expected_data29] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[APJ6JRA9NG5V4-input_data30-expected_data30] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[APJ6JRA9NG5V4-input_data31-expected_data31] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1805IZSGTT6HS-input_data32-expected_data32] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1805IZSGTT6HS-input_data33-expected_data33] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1805IZSGTT6HS-input_data34-expected_data34] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1805IZSGTT6HS-input_data35-expected_data35] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A17E79C6D8DWNP-input_data36-expected_data36] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A17E79C6D8DWNP-input_data37-expected_data37] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A17E79C6D8DWNP-input_data38-expected_data38] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A17E79C6D8DWNP-input_data39-expected_data39] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2NODRKZP88ZB9-input_data40-expected_data40] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2NODRKZP88ZB9-input_data41-expected_data41] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2NODRKZP88ZB9-input_data42-expected_data42] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2NODRKZP88ZB9-input_data43-expected_data43] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A33AVAJ2PDY3EV-input_data44-expected_data44] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A33AVAJ2PDY3EV-input_data45-expected_data45] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A33AVAJ2PDY3EV-input_data46-expected_data46] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A33AVAJ2PDY3EV-input_data47-expected_data47] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1F83G8C2ARO7P-input_data48-expected_data48] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1F83G8C2ARO7P-input_data49-expected_data49] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1F83G8C2ARO7P-input_data50-expected_data50] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1F83G8C2ARO7P-input_data51-expected_data51] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A39IBJ37TRP1C6-input_data52-expected_data52] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A39IBJ37TRP1C6-input_data53-expected_data53] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A39IBJ37TRP1C6-input_data54-expected_data54] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A39IBJ37TRP1C6-input_data55-expected_data55] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1VC38T7YXB528-input_data56-expected_data56] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1VC38T7YXB528-input_data57-expected_data57] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1VC38T7YXB528-input_data58-expected_data58] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1VC38T7YXB528-input_data59-expected_data59] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A19VAU5U5O7RUS-input_data60-expected_data60] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A19VAU5U5O7RUS-input_data61-expected_data61] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A19VAU5U5O7RUS-input_data62-expected_data62] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A19VAU5U5O7RUS-input_data63-expected_data63] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ATVPDKIKX0DER-input_data64-expected_data64] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ATVPDKIKX0DER-input_data65-expected_data65] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ATVPDKIKX0DER-input_data66-expected_data66] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[ATVPDKIKX0DER-input_data67-expected_data67] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2Q3Y263D00KWC-input_data68-expected_data68] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2Q3Y263D00KWC-input_data69-expected_data69] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2Q3Y263D00KWC-input_data70-expected_data70] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2Q3Y263D00KWC-input_data71-expected_data71] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2EUQ1WTGCTBG2-input_data72-expected_data72] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2EUQ1WTGCTBG2-input_data73-expected_data73] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2EUQ1WTGCTBG2-input_data74-expected_data74] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A2EUQ1WTGCTBG2-input_data75-expected_data75] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1AM78C64UM0Y8-input_data76-expected_data76] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1AM78C64UM0Y8-input_data77-expected_data77] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1AM78C64UM0Y8-input_data78-expected_data78] PASSED
unit_tests/test_transform_function.py::test_transform_seller_feedback[A1AM78C64UM0Y8-input_data79-expected_data79] PASSED

============================================================================================== warnings summary ==============================================================================================
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py::test_finance_stream_next_token
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py::test_financial_event_groups_stream_request_params
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py::test_financial_event_groups_stream_parse_response
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py::test_financial_events_stream_request_params
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py::test_financial_events_stream_parse_response
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py::test_streams
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py::test_streams
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py::test_streams
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py::test_streams
  /home/vladimir/.virtualenvs/amzseller/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:42: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator: HttpAuthenticator = NoAuth()

airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_finance_streams.py: 5 warnings
airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py: 5 warnings
  /home/vladimir/.virtualenvs/amzseller/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py::test_streams
  /home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py:55: DeprecationWarning: Call to deprecated class Oauth2Authenticator. (Use airbyte_cdk.sources.streams.http.requests_native_auth.Oauth2Authenticator instead) -- Deprecated since version 0.1.20.
    auth = AWSAuthenticator(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================================================================================== 93 passed, 20 warnings in 0.77s =======================================================================================

```

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
@vladimir-remar thanks for the contribution. Could you help with these things
1.  Run `./gradlew format`
2. Follow these steps for updating connector https://docs.airbyte.com/connector-development/#updating-an-existing-connector

**Comment by vladimir-remar** [comment]
> @vladimir-remar thanks for the contribution. Could you help with these things
> 
> 1. Run `./gradlew format`
> 2. Follow these steps for updating connector https://docs.airbyte.com/connector-development/#updating-an-existing-connector

Hello @harshithmullapudi, this is my format output
```bash
vladimir@vremar:airbyte$ ./gradlew :airbyte-integrations:connectors:source-amazon-seller-partner:airbytePythonFormat
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details
Building all of Airbyte.
/home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors
Type-safe dependency accessors is an incubating feature.

> Task :airbyte-integrations:connectors:source-amazon-seller-partner:checkPython
Using python 3.9.13 from /home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.venv (.venv/bin/python)
Using pip 21.3.1 from /home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.venv/lib/python3.9/site-packages/pip (python 3.9)

> Task :airbyte-integrations:connectors:source-amazon-seller-partner:pipInstall
[python] .venv/bin/python -m pip install pip==21.3.1
         Requirement already satisfied: pip==21.3.1 in ./.venv/lib/python3.9/site-packages (21.3.1)
         WARNING: You are using pip version 21.3.1; however, version 22.1.2 is available.
         You should consider upgrading via the '/home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.venv/bin/python -m pip install --upgrade pip' command.
[python] .venv/bin/python -m pip list --format=columns
         Package           Version
         ----------------- -------
         attrs             21.4.0
         black             22.3.0
         click             8.1.3
         coverage          6.3.1
         flake8            4.0.1
         iniconfig         1.1.1
         isort             5.6.4
         mccabe            0.6.1
         mypy              0.930
         mypy-extensions   0.4.3
         packaging         21.3
         pathspec          0.9.0
         pip               21.3.1
         platformdirs      2.5.2
         pluggy            0.13.1
         py                1.11.0
         pycodestyle       2.8.0
         pyflakes          2.4.0
         pyparsing         3.0.9
         pyproject-flake8  0.0.1a2
         pytest            6.1.2
         setuptools        62.1.0
         toml              0.10.2
         tomli             2.0.1
         typing_extensions 4.2.0
         wheel             0.37.1
         WARNING: You are using pip version 21.3.1; however, version 22.1.2 is available.
         You should consider upgrading via the '/home/vladimir/Work/Airbyte_projects/Fork/airbyte/airbyte-integrations/connectors/source-amazon-seller-partner/.venv/bin/python -m pip install --upgrade pip' command.

> Task :airbyte-integrations:connectors:source-amazon-seller-partner:isortFormat
[python] .venv/bin/python -m isort --settings-file=/home/vladimir/Work/Airbyte_projects/Fork/airbyte/pyproject.toml ./
         Skipped 6 files

> Task :airbyte-integrations:connectors:source-amazon-seller-partner:blackFormat
[python] .venv/bin/python -m black --config /home/vladimir/Work/Airbyte_projects/Fork/airbyte/pyproject.toml ./
         All done! ✨ 🍰 ✨
         14 files left unchanged.

> Task :airbyte-integrations:connectors:source-amazon-seller-partner:flakeCheck
[python] .venv/bin/python -m pflake8 --config /home/vladimir/Work/Airbyte_projects/Fork/airbyte/pyproject.toml ./

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 44s
16 actionable tasks: 9 executed, 7 up-to-date

```

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py [hunk 1]
```diff
@@ -820,3 +820,94 @@ def path(self, **kwargs) -> str:
...
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Mapping[str, Any]:
+
+        # For backwards
+
+        return {"reportId": stream_slice.get("report_id")}
+
+    def stream_slices(
```
[the remark this answers is not in the document]
Comment (harshithmullapudi) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:872]: @vladimir-remar I don't think it's right way to use the stream_slices here. @marcosmarxm do you have any suggestions here?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by harshithmullapudi** [comment]
Hey @vladimir-remar could you help on those changes Marcos requested?

**Comment by vladimir-remar** [comment]
Thanks for your patience, I will check it ASAP.

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py [hunk 2]
```diff
@@ -820,3 +820,94 @@ def path(self, **kwargs) -> str:
...
+        """
+        From https://developer-docs.amazon.com/sp-api/docs/report-type-values
+        documentation:
+        ```Settlement reports cannot be requested or scheduled.
+            They are automatically scheduled by Amazon.
+            You can search for these reports using the getReports operation.
+        ```
+        """
+
+        strict_start_date = pendulum.now("utc").subtract(days=90)
```
[the remark this answers is not in the document]
Comment (vladimir-remar) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:884]: sure, That strict limit comes from documentation:
```
createdSince
The earliest report creation date and time for reports to include in the response, in ISO 8601 date time format. The default is 90 days ago. Reports are retained for a maximum of 90 days.
```
For testing porpuses if a date is less than 90 days ago the status from response is 400.

**Review by harshithmullapudi — APPROVED** [review]

**Pull request merged**